src/invidious/channels/about.cr

```diff
@@ -29,29 +29,14 @@ struct AboutRelatedChannel
 end
 
 def get_about_info(ucid, locale)
-  result = YT_POOL.client &.get("/channel/#{ucid}/about?gl=US&hl=en")
-  if result.status_code != 200
-    result = YT_POOL.client &.get("/user/#{ucid}/about?gl=US&hl=en")
+  begin
+    initdata = request_youtube_api_browse(browse_id: ucid, params: "")
+  rescue
+    raise InfoException.new("Could not get channel info.")
   end
 
-  if md = result.headers["location"]?.try &.match(/\/channel\/(?<ucid>UC[a-zA-Z0-9_-]{22})/)
-    raise ChannelRedirect.new(channel_id: md["ucid"])
-  end
-
-  if result.status_code != 200
-    raise InfoException.new("This channel does not exist.")
-  end
-
-  about = XML.parse_html(result.body)
-  if about.xpath_node(%q(//div[contains(@class, "channel-empty-message")]))
-    raise InfoException.new("This channel does not exist.")
-  end
-
-  initdata = extract_initial_data(result.body)
-  if initdata.empty?
-    error_message = about.xpath_node(%q(//div[@class="yt-alert-content"])).try &.content.strip
-    error_message ||= translate(locale, "Could not get channel info.")
-    raise InfoException.new(error_message)
+  if initdata["alerts"]?.try &.[0]?.try &.["alertRenderer"]?.try &.["type"]? == "ERROR"
```

Comment:
can you please use the `dig?()` method instead of this `.try` chain? (See #2231)

Reply:
Not sure how to use the dig function on an array though.

Reply:
You can even do: `initdata.dig?("alerts", 0, "alertRenderer", "type")

Reply:
Awesome this function is magic.